astropy/io/ascii/tdat.py

```diff
@@ -515,19 +515,18 @@ def write(self, lines):
         lines.append("# Table Parameters")
         lines.append("#")
         for col in self.cols:
-            if str(col.info.dtype) in self._dtype_dict_out:
-                ctype = self._dtype_dict_out[str(col.info.dtype)]
-            elif "int" in str(col.info.dtype):
+            if str(col_type := col.info.dtype) in self._dtype_dict_out:
+                ctype = self._dtype_dict_out[str(col_type)]
+            elif col_type.kind == "i":
                 ctype = "int4"
-            elif "float" in str(col.info.dtype):
+            elif col_type.kind == "f":
                 ctype = "float8"
-            elif "<U" in str(col.info.dtype):
-                ctype = f"char{str(col.info.dtype).rsplit('<U', maxsplit=1)[-1]}"
+            elif col_type.kind == "U":
+                ctype = f"char{str(col_type).rsplit('<U', maxsplit=1)[-1]}"
```

Comment:
This would still go wrong if the dtype is `>U`, right? I suggest instead `f"char{col_type.itemsize // 4}" or `f"char{col_type.str.rpartition("U")[-1]}"`

Reply:
Right; just wanted to check how to best fix the split and then forgot about it – I was not sure if itemsize / 4 would work everywhere, but if it does, that looks like the best option to me, thanks!